google/cloud/ndb/_cache.py

```diff
@@ -645,6 +657,7 @@ def global_unlock_for_write(key, lock):
     Returns:
         tasklets.Future: Eventual result will be :data:`None`.
     """
+    utils.logging_debug(log, "unlock for write: {}", lock)
```

Comment:
The sentance "If no other locks remain, the key will be deleted." in the function docstring needs to be removed.

Reply:
The definition of `new_value` below has always bugged me.  It doesn't actually check if the lock is in the old value, potentially hiding bugs.

consider adding asserions:

```
def new_value(old_value):
    assert old_value and lock in old_value, "Lock was lost"
    return old_value.replace(lock, b"")
```

Reply:
> The sentance "If no other locks remain, the key will be deleted." in the function docstring needs to be removed.

227d19b